ionoscloud/resource_ipfailover.go

```diff
@@ -84,118 +95,89 @@ func resourceLanIPFailoverCreate(ctx context.Context, d *schema.ResourceData, me
 		return diags
 	}
 
-	d.SetId(*lan.Id)
+	d.SetId(uuidgen.ResourceUuid().String())
 
-	return resourceLanIPFailoverRead(ctx, d, meta)
+	return nil
 }
 
 func resourceLanIPFailoverRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceLanIPFailoverUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(services.SdkBundle).CloudApiClient
+	dcId := d.Get("datacenter_id").(string)
+	lanId := d.Get("lan_id").(string)
+	ip := d.Get("ip").(string)
 
-	lan, apiResponse, err := client.LANsApi.DatacentersLansFindById(ctx, d.Get("datacenter_id").(string), d.Id()).Execute()
-	logApiRequestTime(apiResponse)
+	if d.HasChange("nicuuid") {
+		oldValue, newValue := d.GetChange("nicuuid")
+		newNicUuid := newValue.(string)
+		oldNicUuid := oldValue.(string)
 
-	if err != nil {
-		if httpNotFound(apiResponse) {
-			d.SetId("")
-			return nil
+		// First, retrieve the existent IP Failover groups
+		lan, apiResponse, err := lanSvc.FindLanById(*client, ctx, dcId, lanId)
+		if err != nil {
+			return diag.FromErr(err)
 		}
-		diags := diag.FromErr(fmt.Errorf("an error occured while fetching a lan ID %s %w", d.Id(), err))
-		return diags
-	}
 
-	failoverSlice := lan.Properties.IpFailover
-	if lan.Properties != nil && failoverSlice != nil && len(*failoverSlice) > 0 {
-		firstFailover := (*failoverSlice)[0]
-		if firstFailover.Ip != nil {
-			err := d.Set("ip", firstFailover.Ip)
-			if err != nil {
-				diags := diag.FromErr(fmt.Errorf("error while setting ip property for IpFailover %s: %w", d.Id(), err))
-				return diags
-			}
-		}
-		if firstFailover.NicUuid != nil {
-			err := d.Set("nicuuid", firstFailover.NicUuid)
-			if err != nil {
-				diags := diag.FromErr(fmt.Errorf("error while setting nicuuid property for IpFailover %s: %w", d.Id(), err))
-				return diags
-			}
-		}
-	}
+		// Add the new IP failover group to the list
+		*lan.Properties.IpFailover = append(*lan.Properties.IpFailover, ionoscloud.IPFailover{
+			Ip:      &ip,
+			NicUuid: &newNicUuid,
+		})
+
+		// Remove the old IP failover group from the list
+		*lan.Properties.IpFailover = slice.DeleteFrom(*lan.Properties.IpFailover, ionoscloud.IPFailover{
+			Ip:      &ip,
+			NicUuid: &oldNicUuid,
+		})
+
+		properties := &ionoscloud.LanProperties{}
```

Comment:
no need to create a ref here I think

Reply:
@cristiGuranIonos I removed the reference in this commit: 7a35821c28be88119348db39cb88c96db7c113b9